Sources/Shared/Core/Request+EndpointType.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  Request+EndpointType.swift
+//  VimeoNetworking
+//
+//  Created by Rogerio de Paula Assis on 10/3/19.
+//  Copyright © 2019 Vimeo. All rights reserved.
+//
+
+import Foundation
+
+extension Request: EndpointType {
+
+    public var headers: HTTPHeaders? {
+        return nil
+    }
+
+    // TODO: This method should be able to serialize its own parameters however with the current
+    // way the SessionManager is designed we are unable to do that as it would require pretty
+    // substantial refactoring. So for now the request serialization is the responsibility of the
+    // SessionManaging type [RDPA 10/02/2019].
```

Comment:
Note to check in on this part

Reply:
I'd like to leave that TODO there for the future as we are unable to fix this with the way things are currently structured.